integration-tests/src/nft.rs

```diff
@@ -0,0 +1,251 @@
+	assert_eq!(
+		is_allowed,
+		<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_create(&class_type)
+	);
+}
+
+#[test_case(ClassType::Marketplace, ALLOW ; "marketplace class")]
+#[test_case(ClassType::LiquidityMining, NOT_ALLOW ; "liquidity mining class")]
+#[test_case(ClassType::Redeemable, NOT_ALLOW ; "redeemable class")]
+#[test_case(ClassType::Auction, NOT_ALLOW ; "auction class")]
+#[test_case(ClassType::HydraHeads, NOT_ALLOW ; "hydra heads class")]
+fn test_nft_permission_for_minting(class_type: ClassType, is_allowed: bool) {
+	assert_eq!(
+		is_allowed,
+		<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_mint(&class_type)
+	);
+}
+
+#[test_case(ClassType::Marketplace, ALLOW ; "marketplace class")]
+#[test_case(ClassType::LiquidityMining, ALLOW ; "liquidity mining class")]
+#[test_case(ClassType::Redeemable, NOT_ALLOW ; "redeemable class")]
+#[test_case(ClassType::Auction, NOT_ALLOW ; "auction class")]
+#[test_case(ClassType::HydraHeads, NOT_ALLOW ; "hydra heads class")]
+fn test_nft_permission_for_transfer(class_type: ClassType, is_allowed: bool) {
+	assert_eq!(
+		is_allowed,
+		<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_transfer(&class_type)
+	);
+}
+
+#[test_case(ClassType::Marketplace, ALLOW ; "marketplace class")]
+#[test_case(ClassType::LiquidityMining, NOT_ALLOW ; "liquidity mining class")]
+#[test_case(ClassType::Redeemable, NOT_ALLOW ; "redeemable class")]
+#[test_case(ClassType::Auction, NOT_ALLOW ; "auction class")]
+#[test_case(ClassType::HydraHeads, NOT_ALLOW ; "hydra heads class")]
+fn test_nft_permission_for_burning(class_type: ClassType, is_allowed: bool) {
+	assert_eq!(
+		is_allowed,
+		<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_burn(&class_type)
+	);
+}
+
+#[test_case(ClassType::Marketplace, ALLOW ; "marketplace class")]
+#[test_case(ClassType::LiquidityMining, NOT_ALLOW ; "liquidity mining class")]
+#[test_case(ClassType::Redeemable, NOT_ALLOW ; "redeemable class")]
+#[test_case(ClassType::Auction, NOT_ALLOW ; "auction class")]
+#[test_case(ClassType::HydraHeads, NOT_ALLOW ; "hydra heads class")]
+fn test_nft_permission_for_class_destroying(class_type: ClassType, is_allowed: bool) {
+	assert_eq!(
+		is_allowed,
+		<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::can_destroy(&class_type)
+	);
+}
+
+#[test_case(ClassType::Marketplace, ALLOW ; "marketplace class")]
+#[test_case(ClassType::LiquidityMining, NOT_ALLOW ; "liquidity mining class")]
+#[test_case(ClassType::Redeemable, NOT_ALLOW ; "redeemable class")]
+#[test_case(ClassType::Auction, NOT_ALLOW ; "auction class")]
+#[test_case(ClassType::HydraHeads, NOT_ALLOW ; "hydra heads class")]
+fn test_nft_permission_for_deposit(class_type: ClassType, is_allowed: bool) {
+	assert_eq!(
+		is_allowed,
+		<basilisk_runtime::Runtime as pallet_nft::Config>::Permissions::has_deposit(&class_type)
+	);
+}
+
+// Nonfungibles traits
```

Comment:
hm, now when I'm looking at it you were right. This look more like unit tests, tested code is not using permissions at all. I should look at nft implementation first, sorry.
I think they should be in warehosue as you said, but I'm ok if it's stay here.